### PR TITLE
Update terraform for S3 bucket to reflect new access control restrictions

### PR DESCRIPTION
**Pull request by sjockers**
Fix problems with the provisioning of the S3 bucket, due to a recent change in AWS related to `acl="public-read"`.

**Background:** As of April 2023, creating a new S3 bucket with the Access Control List (ACL) option [is no longer possible](https://aws.amazon.com/about-aws/whats-new/2022/12/amazon-s3-automatically-enable-block-public-access-disable-access-control-lists-buckets-april-2023/). Due to this change by AWS, the S3 provisioning in this repo is broken. 

See also https://github.com/technologiestiftung/giessdenkiez-de-aws-s3-terraform/issues/6 (different repo, same issue)

---

Bonus fixes:
- [Remove hardcoded tfvars config, add example config instead](https://github.com/technologiestiftung/giessdenkiez-de-dwd-harvester/commit/660fb1556966764349cdbf92c403241d5495d499)
- [Add asdf tool-versions config](https://github.com/technologiestiftung/giessdenkiez-de-dwd-harvester/commit/328ba45307fff7843a3c9eb779fa364461dc135b)

### CODE REVIEW

File: terraform/.tool-versions
Comment (ff6347) [inline terraform/.tool-versions:1]: The version is pretty outdated. Current version is [1.4.6](https://developer.hashicorp.com/terraform/downloads?product_intent=terraform)

**Review by ff6347 — CHANGES_REQUESTED** [review]
Thank you for you contribution. Only thing I see is that v0.15.1 of terraform cli is outdated. Could you try it with the latest version?

**Comment by sjockers** [comment]
Updated terraform to 1.4.6, which still works fine.

**Review by ff6347 — APPROVED** [review]

**Comment by ff6347** [comment]
@all-contributors please add @sjockers for infra, code and bugs

**Pull request merged**